lib/auth/MatrixAuthProvider.js

```diff
@@ -112,21 +112,22 @@ class MatrixAuthProvider {
      * @param {string} type - The type of the room or space.
      * @param {string} template - The template for the room or space.
      * @param {string} parentId - The ID of the parent space, if any.
+     * @TODO: Missing some @param declarations
      * @returns {Promise<Object>} A promise that resolves to the created room or space.
      * @throws {Error} If there was an error creating the room or space.
      */
-    async createRoom(name, isSpace, topic, joinRule, type, template, parentId) {
+    async createRoom(name, isSpace, topic, joinRule, type, template, parentId, application, visibility, historyVisible, encrypted) {
```

Comment:
Where are we making use of any of these new parameters?
If we don't use them, let's not add them.

I guess this is a leftover from some previous concepts?